### PR TITLE
[mypy] Add some variable annotations

### PR DESCRIPTION
**Pull request by rodrigc**
https://twistedmatrix.com/trac/ticket/9870

This adds some annotations to reduce mypy errors.

### CODE REVIEW

File: src/twisted/test/test_process.py [hunk 1]
```diff
@@ -389,7 +390,7 @@ class UtilityProcessProtocol(protocol.ProcessProtocol):
 
     @ivar programName: The name of the program to run.
     """
-    programName = None
+    programName = None  # type: Optional[bytes]
```
Comment (adiroiban) [inline src/twisted/test/test_process.py]: since this is testing code, it can be initialized with a default value.

I don't think that having programName as None is a valid value as the code should always fail in that case.

```suggestion
    programName = b'define-your-own-name'
```

File: src/twisted/web/http.py [hunk 2]
```diff
@@ -733,8 +733,8 @@ class Request:
     finished = 0
     code = OK
     code_message = RESPONSES[OK]
-    method = "(no method yet)"
-    clientproto = b"(no clientproto yet)"
+    method = b"(no method yet)"  # type: bytes
```
Comment (adiroiban) [inline src/twisted/web/http.py:736]: why is this required? should this be auto detected from the normal assignment ?
Is this a mypy bug?

**Review by adiroiban — APPROVED** [review]
Looks good. Only minor comments. Thanks

File: src/twisted/web/http.py (hunk 2 above)
Comment (rodrigc) [reply src/twisted/web/http.py:736]: All subclasses define method as bytes, which is correct.  It is incorrectly defined as str here.

For example, elsewhere in this file:

```
        if self.method == b"POST" and ctype and clength:
            mfd = b'multipart/form-data'
            key, pdict = _parseHeader(ctype)
```

File: src/twisted/test/test_process.py (hunk 1 above)
Comment (rodrigc) [reply src/twisted/test/test_process.py]: Done
